### PR TITLE
feat(cli): use `native-tls-vendored` on mobile instead of `rustls-tls`

### PR DESCRIPTION
**Pull request by lucasfernog**
### What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Docs
- [ ] New Binding issue #___
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

### Does this PR introduce a breaking change?

- [ ] Yes, and the changes were approved in issue #___
- [ ] No

### Checklist
- [ ] When resolving issues, they are referenced in the PR's title (e.g `fix: remove a typo, closes #___, #___`)
- [x] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/tauri/blob/dev/.changes/readme.md).
- [ ] I have added a convincing reason for adding this feature, if necessary

### CODE REVIEW

File: tooling/cli/src/interface/rust.rs [hunk 1]
```diff
@@ -349,19 +349,15 @@ fn shared_options(
...
-    }
+  let all_features = app_settings
+    .manifest
+    .all_enabled_features(if let Some(f) = features { f } else { &[] });
+  if !all_features.contains(&"tauri/rustls-tls".into()) {
+    features.get_or_insert(Vec::new()).push(if mobile {
+      "tauri/native-tls-vendored".into()
+    } else {
+      "tauri/native-tls".into()
+    });
```
Comment (amrbashir) [inline tooling/cli/src/interface/rust.rs:360]: do we really need to manage these features on the CLI? can't we just let the users activate the feature he needs? 
and in the CLI template we should probably be using:
```toml
[target."cfg(not(any(\"android\", \"ios\")))".dependencies]
tauri = { version = "2.0.0-alpha.9", features = []}

[target."cfg(any(\"android\", \"ios\"))".dependencies]
tauri = { version = "2.0.0-alpha.9", features = ["native-tls-vendored"]}
```

Comment (lucasfernog) [reply tooling/cli/src/interface/rust.rs:360]: That has more implications, like the manifest rewrite we currently have (we automatically enable "system-tray, macos-private-api, isolation and protocol-asset). This duplication is also ugly :(

Comment (amrbashir) [reply tooling/cli/src/interface/rust.rs:360]: yeah, it is probably fine to keep in the CLI, but maybe only apply the feature flag if none of the 3 flags is used?

Comment (lucasfernog) [reply tooling/cli/src/interface/rust.rs:360]: We actually only need to check for the rustls-tls feature. Enabling `native-tls` when the vendored flag is enabled does not change anything.

**Comment by lucasfernog** [comment]
And seems like we'll need to stick with rustls-tls, Windows still doesn't work because or Perl.

**Comment by amrbashir** [comment]
lets keep it as is for now, is there any disadvantage to using `rustls-tls`?

**Comment by lucasfernog** [comment]
Not yet. But the lack of option can bite us in the future.

**Comment by lucasfernog** [comment]
It would be nice if we could use `native-tls-vendored` because it would mean less feature flag management on the plugin side, see the [API example](https://github.com/tauri-apps/plugins-workspace/pull/375).

**Comment by amrbashir** [comment]
I wasted so much time in the past on openssl/perl issue with Windows with no avail.

**Comment by lucasfernog** [comment]
Alright, let's punt this again.

**Comment by wusyong** [comment]
@amrbashir I think PR is because of https://github.com/tauri-apps/plugins-workspace/pull/375
My concern is Rustls uses ring which is pure assembly. It doesn't support many platform you would imagine.
Did you confirm it rustls can work on most emulator and device platform? (like 32/64 bit, x86-64/arm, apple/android etc)

**Comment by amrbashir** [comment]
I have been using tauri on emulator and real android devices, and I never had any issues (assuming tauri was always using `rustls-tls`)

**Comment by wusyong** [comment]
Alright if you both have no problem with it then let's close this one.
I'll review the PR in plugin workspace later.

**Pull request closed without merging**